assets/wpuf/js/frontend-form.js

```diff
@@ -717,7 +717,7 @@
         },
 
         isValidURL: function(url) {
-            var urlregex = new RegExp("^(http:\/\/www.|https:\/\/www.|ftp:\/\/www.|www.|http:\/\/|https:\/\/){1}([0-9A-Za-z]+\.)");
+            var urlregex = new RegExp(/^(?:(?:https?|ftp):\/\/)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/\S*)?$/);

```

Comment:
What was wrong with the old regex / where did you get the new regex?

Does `<input type="url">` ([link](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/url)) not validate the url correctly?

Reply:
`<input type="url">` only makes sure there is a protocol in the field i.e. http:// , https://, or ftp://. It still allows you to put something like http://test.c which is not a valid URL. The regex I added came from https://gist.github.com/dperini/729294, so that the URLs that are added are truly being validated. The old regex was just doing what `<input type="url">`  was doing which forced a protocol.

Reply:
Can you put a comment above that so we know? IE

```js

/**
 * Validate a url.
 *
 * While we could use the standard HTML5 form validator for urls, this regex does a more thorough job.
 *
 * @link https://gist.github.com/dperini/729294
 */
isValidURL: function(url) {
    var urlregex = new RegExp(/^(?:(?:https?|ftp):\/\/)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/\S*)?$/);<!--EndFragment-->
```

Reply:
Description   comment added